frappe/integration_broker/oauth2.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import unicode_literals
+import frappe, json
+from frappe.oauth import OAuthWebRequestValidator, WebApplicationServer
+from oauthlib.oauth2 import FatalClientError, OAuth2Error
+from urllib import quote, urlencode
+from frappe.integrations.doctype.oauth_provider_settings.oauth_provider_settings import get_oauth_settings
+
+#Variables required across requests
```

Comment:
A good place to add some docs:

See

https://github.com/frappe/frappe/blob/develop/frappe/integrations/doctype/razorpay_settings/razorpay_settings.py#L12